src/index.native.scroll.js

```diff
@@ -105,13 +114,24 @@ class SwipeableViews extends Component {
       index,
     } = nextProps;
 
-    if (typeof index === 'number' && index !== this.props.index) {
+    const indexHasChanged = (typeof index === 'number' && index !== this.props.index);
+
+    if (indexHasChanged) {
       this.setState({
         indexLatest: index,
         offset: {
           x: this.state.viewWidth * index,
         },
       }, () => {
+        if (indexHasChanged) {
```

Comment:
Yep good point, as it's already within the if. This is what you get for coding late in the day😅